package-res/resources/web/pentaho/data/filter/AbstractPropertyFilter.js

```diff
@@ -0,0 +1,95 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "./AbstractFilter",
+  "./_toSpec"
+], function(AbstractFilter, _toSpec) {
+  "use strict";
+
+  /**
+   * @name AbstractPropertyFilter
+   * @memberOf pentaho.data.filter
+   * @class
+   * @extends pentaho.data.filter.AbstractFilter
+   * @abstract
+   * @amd pentaho/data/filter/AbstractPropertyFilter
+   *
+   * @classdesc A base class for filters that represent a set of elements which
+   * contain a given property
+   * and whose value matches against a reference value.
+   *
+   * @description Creates an instance that filters a data set by comparing the value of property to a reference value.
+   * @param {string} property The name of the property.
+   * @param {any} value - The reference value.
+   *
+   */
+  var AbstractPropertyFilter = AbstractFilter.extend("pentaho.data.filter.AbstractPropertyFilter", /** @lends pentaho.data.filter.AbstractPropertyFilter# */{
+    constructor: function(property, value) {
+      this.value = value;
+      this.property = property;
+      Object.freeze(this);
+    },
+
+    /**
+     * The name of the property containing the value used for filtering.
+     * @type {string}
+     * @readonly
+     */
+    property: null,
+
+    /**
+     * Reference value of the [property]{@link pentaho.data.filter.AbstractFilter#property} used for filtering.
+     * @type {any}
+     * @readonly
+     */
+    value: null,
```

Comment:
I now see why it is described as "reference value". This is not correct in my view. The value being fixed or not, or a single one or not, depends on the actual operator. So this is not abstract behavior.